app/controllers/coronavirus_form/business_details_controller.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+class CoronavirusForm::BusinessDetailsController < ApplicationController
+  include ActionView::Helpers::SanitizeHelper
+  include FieldValidationHelper
+
+  REQUIRED_FIELDS = %w(company_name).freeze
+
+  def show
+    render "coronavirus_form/#{PAGE}"
+  end
+
+  def submit
+    @business_details = sanitized_business_details(params)
+    session[:business_details] = @business_details
+
+    invalid_fields = validate_fields(@business_details)
+
+    if invalid_fields.any?
+      flash.now[:validation] = invalid_fields
+      render "coronavirus_form/#{PAGE}"
+    else
+      redirect_to controller: session["check_answers_seen"] ? "coronavirus_form/check_answers" : "coronavirus_form/#{NEXT_PAGE}", action: "show"
+    end
+  end
+
+private
+
+  NEXT_PAGE = "contact_details"
+  PAGE = "business_details"
+
+  def validate_fields(business_details)
+    missing_fields = validate_missing_fields(business_details)
+    company_size_validation = validate_radio_field("#{PAGE}.company_size", radio: business_details["company_size"])
+    company_location_validation = validate_radio_field("#{PAGE}.company_location", radio: business_details["company_location"])
+    missing_fields + company_size_validation + company_location_validation
+  end
+
+  def validate_missing_fields(business_details)
```

Comment:
is it worth using the helper https://github.com/alphagov/govuk-coronavirus-business-volunteer-form/blob/master/app/helpers/field_validation_helper.rb#L4 instead?

Reply:
Good spot.  We're not actually using `validate_mandatory_text_fields` anywhere 🤔 .